### PR TITLE
Moving to React and Webpack

### PR DESCRIPTION
**Pull request by mohamed-moanis**
Fixes # 22

Changes proposed in this pull request:
- Use React as UI framework
- Use Webpack to bundle JS files
- Restructure the project directoy
- Add travis-ci config file for NodeJs
- Add unit tests using Jest and enzyme.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ function createWindow () {
 
   // and load the index.html of the app.
   win.loadURL(url.format({
-    pathname: path.join(__dirname, '../views/main-window.html'),
+    pathname: path.join(__dirname, 'app/index.html'),
```
Comment (Venkat5694) [inline main.js]: Move index.html file to public folder

File: package.json [hunk 2]
```diff
@@ -16,9 +16,18 @@
   "copyright": "© 2018, Deer",
   "private": true,
   "scripts": {
+    "dev": "webpack",
     "start": "electron ."
   },
   "devDependencies": {
-    "electron": "~1.7.8"
+    "electron": "~1.7.8",
```
Comment (Venkat5694) [inline package.json]: Electron is not a dev dependency.

File: app/App.jsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import React, {Component} from 'react'
+import {render} from 'react-dom'
+
+export default class App extends Component {
```
Comment (Venkat5694) [inline app/App.jsx]: Please ensure that all new/modified components have corresponding tests.
Use [Jest](https://github.com/facebook/jest) for writing tests.

Comment (mohamed-moanis) [reply app/App.jsx]: Okay, I used Jest and enzyme as documented 
https://facebook.github.io/jest/docs/en/tutorial-react.html

**Comment by mohamed-moanis** [comment]
All required changes are done @Venkat5694 
Ready for review. Please tell me if Jest is configured correctly, because I wasn't sure that this is the best practice with babel and webpack.

File: app/entry.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import React from 'react'
+import {render} from 'react-dom'
+import App from './App.jsx'
```
Comment (Venkat5694) [inline app/entry.js]: JS module imports do not require the .js or .jsx extensions

File: package.json [hunk 5]
```diff
@@ -16,9 +16,25 @@
   "copyright": "© 2018, Deer",
   "private": true,
   "scripts": {
-    "start": "electron ."
+    "dev": "webpack",
```
Comment (Venkat5694) [inline package.json]: This is the build step. Let's call it build instead of dev.

File: app/entry.js (hunk 4 above)
Comment (mohamed-moanis) [reply app/entry.js]: This needs an update in webpack configuration to resolve file extensions

**Comment by mohamed-moanis** [comment]
Oh, I didn't notice the assets typo. Typo from master branch,

**Comment by Venkat5694** [comment]
Great work @mohamed-moanis.

**Pull request merged**